Oracle.md

```diff
@@ -0,0 +1,194 @@
+This necessary information is committed in a so-called [_event descriptor_](#Event-descriptor) that will be further detailed in this document.
+
+## Table of Contents
+
+- [Event descriptor](#event-descriptor)
+   - [Simple enumeration](#simple-enumeration)
+      - [Example: Weather tomorrow](#example-weather-tomorrow)
+   - [Range](#range)
+      - [Example: tomorrow's temperature](#example-tomorrows-temperature)
+   - [Large range](#large-range)
+      - [Example: BTC/USD rate](#example-btcusd-rate)
+   - [Serialization and signing of outcome values](#serialization-and-signing-of-outcome-values)
+   - [Serialization of event descriptors](#serialization-of-event-descriptors)
+      - [Version 0 `enum_event_descriptor`](#version-0-enum_event_descriptor)
+      - [Version 0 `range_event_descriptor`](#version-0-range_event_descriptor)
+      - [Version 0 `large_range_event_descriptor`](#version-0-large_range_event_descriptor)
+   - [Oracle events](#oracle-events)
+      - [Version 0 `oracle_event`](#version-0-oracle_event)
+- [Oracle announcements](#oracle-announcements)
+      - [Version 0 `oracle_announcement`](#version-0-oracle_announcement)
+
+
+
+## Event descriptor
+
+An event descriptor provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Four kinds of event outcomes are defined (note that a single event can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the outcomes can simply be enumerated.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- count: the number of possible outcomes
+- step: the increment
+- unit: the unit of the outcome value
+- precision: the precision of the outcome representing the base 10 exponent by which to multiply the signed number to obtain the actual outcome value.
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+count: 201
+step: 1
+unit: °C
+precision: 0
+```
+
+### Large range
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using numerical decomposition.
+
+The event descriptor should include:
+
+- base: a number representing the base in which the outcome value is decomposed and which will indicate the possible range of each digit that will be signed (e.g. 2 for binary, 10 for decimal, 16 for hex-decimal).
+- is-signed: a boolean value indicating whether the outcomes can be negative.
```

Comment:
Hmm. So I'm suspicious of this. I commented this on our implementation PR 

https://github.com/bitcoin-s/bitcoin-s/pull/2185#discussion_r506662249

What does `is_signed` mean? Maybe I am over complicating this but there are numerous signed number representations. 

https://en.wikipedia.org/wiki/Signed_number_representations

Which one are we using if `is_signed` is true?

Reply:
Looking at this further, it seems that this scheme is implicitly using [one's complement](https://en.wikipedia.org/wiki/Ones%27_complement). 

If I understand this correctly, we can have both `-0` and `+0`. I guess this means an oracle needs to sign both of these representations?

Reply:
No, it is currently defined to sign "+" for zero

Reply:
> an additional R-value must be provided as the first element in this array, which will be used to sign the string "+" in case of an outcome with a positive value or zero value and the string "-" in case of an outcome with a negative one.